app/src/main/java/com/theuhooi/uhooipicbook/modules/monsterlist/MonsterListRecyclerViewAdapter.kt

```diff
@@ -0,0 +1,59 @@
+package com.theuhooi.uhooipicbook.modules.monsterlist
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import coil.api.load
+import com.theuhooi.uhooipicbook.R
+import com.theuhooi.uhooipicbook.modules.monsterlist.MonsterListFragment.OnListFragmentInteractionListener
+import com.theuhooi.uhooipicbook.modules.monsterlist.entity.MonsterContent.MonsterItem
+import kotlinx.android.synthetic.main.item_monster_list.view.*
+
+class MonsterListRecyclerViewAdapter(
+    private val monsters: List<MonsterItem>,
+    private val listener: OnListFragmentInteractionListener?
+) : RecyclerView.Adapter<MonsterListRecyclerViewAdapter.ViewHolder>() {
+
+    // MARK: Stored Instance Properties
+
+    private val onClickListener: View.OnClickListener
+
+    // MARK: Initializers
+
+    init {
+        onClickListener = View.OnClickListener { v ->
+            val item = v.tag as MonsterItem
+            this.listener?.onListFragmentInteraction(item)
+        }
+    }
+
+    // MARK: View Life-Cycle Methods
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val view = LayoutInflater.from(parent.context)
+            .inflate(R.layout.item_monster_list, parent, false)
+        return ViewHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        val item = this.monsters[position]
+        holder.iconImageView.load(item.iconUrlString)
+        holder.nameTextView.text = item.name
+
+        holder.view.tag = item
+        holder.view.setOnClickListener(onClickListener)
+    }
+
+    override fun getItemCount(): Int = this.monsters.size
+
+    // MARK: Inner Classes
+
+    inner class ViewHolder(val view: View) : RecyclerView.ViewHolder(view) {
```

Comment:
This class needs not be `inner`.

Inner classes have internally wrapping class ( in this case `MonsterListRecyclerViewAdapter`), so classes which don't require access to wrapping class should be non-inner class.

Reply:
So that's it.
It was automatically generated, I thought it was good😅
Independent ViewHolder.

Reply:
I meant "non-inner class" as nested classes without `inner` modifier, so declaring the view holder class in adapter class is OK, moreover,  you need not to separate view holder file because a small view holder class is typically defined in adapter class.

see also: https://kotlinlang.org/docs/reference/nested-classes.html

Reply:
Oh.
I understand.
Fix it.